### PR TITLE
Add support for 4232HA

### LINKED ISSUE #76: Support for FT4232HA
Hey!

Is there any chance we could add support for the FT4232HA? It's a different part number (PID) from the 4232H, but I think is API-compatible, just being the automotive variant. I've tried some “brute force” approaches to instantiate it, with no luck. It doesn't seem to report a serial number at all on Linux, nor does it show up with trying to list devices (I get all four ports, but with garbage details).

Thanks!

### PR DESCRIPTION
**Pull request by kiranshila**
Fixes a few typos along the way as well. Fixes #76

Note: This does not add EEPROM support as there appears to be some missing types in the ffi crate, perhaps it needs a bump?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
```
Comment (newAM) [inline CHANGELOG.md:7]:
```suggestion
## [Unreleased]
```

File: src/lib.rs [hunk 2]
```diff
@@ -427,6 +427,7 @@ pub fn list_devices_fs() -> io::Result<Vec<DeviceInfo>> {
             let port_letters: Option<&'static [char]> = match device_type {
                 DeviceType::FT2232H => Some(&['A', 'B']),
                 DeviceType::FT4232H => Some(&['A', 'B', 'C', 'D']),
+                DeviceType::FT4232HA => Some(&['A', 'B', 'C', 'D']),
```
Comment (newAM) [inline src/lib.rs:430]:
```suggestion
                DeviceType::FT4232H | DeviceType::FT4232HA => Some(&['A', 'B', 'C', 'D']),
```

**Review by newAM — APPROVED** [review]
Thanks for adding this!  Just a couple minor suggestions.

Would you like me to make a release after this is merged?

**Comment by kiranshila** [comment]
> Thanks for adding this! Just a couple minor suggestions.

Thanks for this! I'll give this one last squash.

> Would you like me to make a release after this is merged?

Please and thank you, although up to you to decide if you wanted to wait for the full implementation with the EEPROM stuff, but again I think that's something upstream that might take some time to look at. FWIW I tested this on hardware and everything seems to work.

**Review by newAM — APPROVED** [review]

**Pull request merged**